addons/microsoft_calendar/models/calendar.py

```diff
@@ -171,7 +171,7 @@ def write(self, values):
                 self._forbid_recurrence_update()
 
         # When changing the organizer, check its sync status and verify if the user is listed as attendee.
-        if values.get('user_id') and self.user_id.id != values['user_id']:
+        if self and values.get('user_id') and self.user_id.id != values['user_id']:
```

Comment:
Is there any write where you don't have self set?

Reply:
Can't we do an early return if self is empty? I guess calling this without a self won't cause any side effects